x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -180,11 +196,22 @@ export function isModelPlotChartableForDetector(job: Job, detectorIndex: number)
 // Returns a reason to indicate why the job configuration is not supported
 // if the result is undefined, that means the single metric job should be viewable
 export function getSingleMetricViewerJobErrorMessage(job: CombinedJob): string | undefined {
-  // if job has runtime mappings with no model plot
-  if (hasRuntimeMappings(job) && !job.model_plot_config?.enabled) {
-    return i18n.translate('xpack.ml.timeSeriesJob.jobWithRunTimeMessage', {
-      defaultMessage: 'the datafeed contains runtime fields and model plot is disabled',
-    });
+  // if job has at least one composite source that is not terms or date_histogram
+  const aggs = getDatafeedAggregations(job.datafeed_config);
+  if (aggs !== undefined) {
+    const aggBucketsName = getFirstKeyInObject(aggs);
+    if (aggBucketsName !== undefined && aggs[aggBucketsName] !== undefined) {
+      // if fieldName is a aggregated field under nested terms using bucket_script
+
+      if (!hasValidComposite(aggs[aggBucketsName])) {
+        return i18n.translate(
+          'xpack.ml.timeSeriesJob.jobWithUnsupportedCompositeAggregationMessage',
+          {
+            defaultMessage: 'the datafeed contains unsupported composite sources',
```

Comment:
Where does this show up? Should the first char be capitalized and the sentence end with a dot?

Reply:
Originally, the full message will show up like this. But I realized the extra message is redundant so I removed it here [`058c5dd` (#96348)](https://github.com/elastic/kibana/pull/96348/commits/058c5dd196b78806255898d42c3b06638a0e323b).

<img width="292" alt="Screen Shot 2021-04-12 at 10 26 50" src="https://user-images.githubusercontent.com/43350163/114420125-a8300700-9b79-11eb-9e54-0c18322e9677.png">